crates/router/src/connector/bambora/transformers.rs

```diff
@@ -0,0 +1,453 @@
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct AdjustedBy {
+    id: i32,
+    #[serde(rename = "type")]
+    adjusted_by_type: String,
+    approval: i32,
+    message: String,
+    amount: f32,
+    created: String,
+    url: String,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct Links {
+    rel: String,
+    href: String,
+    method: String,
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum PaymentMethod {
+    #[default]
+    Card,
+    Token,
+    PaymentProfile,
+    Cash,
+    Cheque,
+    Interac,
+    ApplePay,
+    AndroidPay,
+    #[serde(rename = "3d_secure")]
+    ThreeDSecure,
+    ProcessorToken,
+}
+
+// Capture
+#[derive(Default, Debug, Clone, Serialize, PartialEq)]
+pub struct BamboraPaymentsCaptureRequest {
+    amount: Option<i64>,
+    payment_method: PaymentMethod,
+}
+
+impl TryFrom<&types::PaymentsCaptureRouterData> for BamboraPaymentsCaptureRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCaptureRouterData) -> Result<Self, Self::Error> {
+        Ok(Self {
+            amount: item.request.amount_to_capture,
+            payment_method: PaymentMethod::Card,
+        })
+    }
+}
+
+// REFUND :
+// Type definition for RefundRequest
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct BamboraRefundRequest {
+    amount: i64,
+}
+
+impl<F> TryFrom<&types::RefundsRouterData<F>> for BamboraRefundRequest {
+    type Error = error_stack::Report<errors::ParsingError>;
+    fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
+        Ok(Self {
+            amount: item.request.amount,
+        })
+    }
+}
+
+// Type definition for Refund Response
+#[allow(dead_code)]
+#[derive(Debug, Serialize, Default, Deserialize, Clone)]
+pub enum RefundStatus {
+    Succeeded,
+    Failed,
+    #[default]
+    Processing,
+}
+
+impl From<RefundStatus> for enums::RefundStatus {
+    fn from(item: RefundStatus) -> Self {
+        match item {
+            RefundStatus::Succeeded => Self::Success,
+            RefundStatus::Failed => Self::Failure,
+            RefundStatus::Processing => Self::Pending,
+        }
+    }
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct RefundResponse {
+    #[serde(deserialize_with = "str_or_i32")]
+    id: String,
+    authorizing_merchant_id: i32,
+    #[serde(deserialize_with = "str_or_i32")]
+    approved: String,
+    #[serde(deserialize_with = "str_or_i32")]
+    message_id: String,
+    message: String,
+    auth_code: String,
+    created: String,
+    amount: f32,
+    order_number: String,
+    #[serde(rename = "type")]
+    payment_type: String,
+    comments: Option<String>,
+    batch_number: Option<String>,
+    total_refunds: Option<f32>,
+    total_completions: Option<f32>,
+    payment_method: String,
+    card: CardData,
```

Comment:
Are we getting payment related information in refund response?

Reply:
yes we get the masked info in response